### PR TITLE
small improvements for tut1.rst

### PR DESCRIPTION
**Pull request by narimiran**

### CODE REVIEW

File: doc/tut1.rst [hunk 1]
```diff
@@ -644,9 +651,9 @@ the exit.
         return
       result = result + i
 
-  echo sumTillNegative() # echos 0
-  echo sumTillNegative(3, 4, 5) # echos 12
-  echo sumTillNegative(3, 4 , -1 , 6) # echos 7
+  echo sumTillNegative() # echoes 0
```
Comment (timotheecour) [inline doc/tut1.rst:647]:
```suggestion
  assert sumTillNegative() == 0
```
simpler + more idiomatic, ditto below

File: doc/tut1.rst [hunk 2]
```diff
@@ -260,6 +234,37 @@ and put it into a data section":
...
+  x = 42        # changes `x` to 42 without changing `y`
+  echo "x ", x  # outputs "x 42"
+  echo "y ", y  # outputs "y 3"
+
+Note that declaring multiple variables with a single assignment that calls a
+procedure can have unexpected results: the compiler will *unroll* the
+assignments and end up calling the procedure several times. If the result of
+the procedure depends on side effects, your variables may end up having
+different values! For safety use side-effect-free procedures if making multiple
+assignments.
```
Comment (Araq) [inline doc/tut1.rst]: Remove this paragraph, it doesn't belong to the basic tutorial.

File: doc/tut1.rst (hunk 1 above)
Comment (Araq) [reply doc/tut1.rst:647]: But then the reader must already know about `assert`.

Comment (timotheecour) [reply doc/tut1.rst:647]: assert is pervasive in programming languages though, it's reasonable to expect users would get its meaning (but the tutorial can add a comment on the 1st instance of `assert`)

Comment (narimiran) [reply doc/tut1.rst:647]: I would agree with this change if this were some test in the stdlib.
This being a tutorial, I prefer `echo` in all the examples where is currently.

File: doc/tut1.rst [hunk 3]
```diff
@@ -793,14 +797,14 @@ Nim provides the ability to overload procedures similar to C++:
 (Note that `toString` is usually the `$ <dollars.html>`_ operator in
 Nim.) The compiler chooses the most appropriate proc for the `toString`
 calls. How this overloading resolution algorithm works exactly is not
-discussed here (it will be specified in the manual soon).  However, it does
-not lead to nasty surprises and is based on a quite simple unification
-algorithm. Ambiguous calls are reported as errors.
+discussed here. However, it does not lead to nasty surprises and is based
+on a quite simple unification algorithm. Ambiguous calls are reported as errors.
```
Comment (Araq) [inline doc/tut1.rst]:
```suggestion
discussed here -- see the manual for details. Ambiguous calls are reported as errors.
```

**Pull request merged**